### ISSUE TITLE
“make test” failed on RHEL 7.0

### ISSUE THREAD

**Issue by NanXiao** [libuv/libuv#501]
Use the newest `v1.x` branch, and “`make test`” failed on `RHEL 7.0`:  

~~~~
make[1]: Entering directory `/home/nan/libuv'
[%   0|+   0|-   0|T   0|S   0]: platform_output
Output from process `platform_output`:
uv_get_process_title: /home/nan/libuv/test/.libs/lt-run-tests
uv_cwd: /home/nan/libuv
uv_resident_set_memory: 778240
uv_uptime: 4025.000000
uv_getrusage:
  user: 0 sec 2429 microsec
  system: 0 sec 3644 microsec
uv_cpu_info:
  model: Intel(R) Core(TM) i5-4310U CPU @ 2.00GHz
  speed: 2594
  times.sys: 1272500
  times.user: 6345200
  times.idle: 30302300
  times.irq: 1300
  times.nice: 6900
  model: Intel(R) Core(TM) i5-4310U CPU @ 2.00GHz
  speed: 2594
  times.sys: 1030900
  times.user: 4833000
  times.idle: 32717200
  times.irq: 0
  times.nice: 6900
  model: Intel(R) Core(TM) i5-4310U CPU @ 2.00GHz
  speed: 2594
  times.sys: 990200
  times.user: 4521100
  times.idle: 33124800
  times.irq: 0
  times.nice: 5300
  model: Intel(R) Core(TM) i5-4310U CPU @ 2.00GHz
  speed: 2594
  times.sys: 910000
  times.user: 5148900
  times.idle: 32086000
  times.irq: 0
  times.nice: 7000
uv_interface_addresses:
  name: lo
  internal: 1
  physical address: 00:00:00:00:00:00
  address: 127.0.0.1
  netmask: 255.0.0.0
  name: enp0s3
  internal: 0
  physical address: 08:00:27:7d:e3:ea
  address: 16.187.246.152
  netmask: 255.255.252.0
  name: virbr0
  internal: 0
  physical address: fe:54:00:d1:7f:7e
  address: 192.168.122.1
  netmask: 255.255.255.0
  name: lo
  internal: 1
  physical address: 00:00:00:00:00:00
  address: ::1
  netmask: ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff
  name: enp0s3
  internal: 0
  physical address: 08:00:27:7d:e3:ea
  address: fe80::a00:27ff:fe7d:e3ea
  netmask: ffff:ffff:ffff:ffff::
  name: virbr0
  internal: 0
  physical address: fe:54:00:d1:7f:7e
  address: fe80::fc54:ff:fedf:fbb7
  netmask: ffff:ffff:ffff:ffff::
  name: vnet0
  internal: 0
  physical address: fe:54:00:d1:7f:7e
  address: fe80::fc54:ff:fed1:7f7e
  netmask: ffff:ffff:ffff:ffff::
=============================================================
[%  80|+ 221|-   0|T   0|S   0]: fs_async_dir
`fs_async_dir` failed: exit code 134
Output from process `fs_async_dir`:
Assertion failed in test/test-fs.c on line 440: dent.type == UV_DIRENT_FILE
=============================================================
[%  82|+ 224|-   1|T   0|S   0]: fs_access
`fs_access` failed: exit code 134
Output from process `fs_access`:
Assertion failed in test/test-fs.c on line 1204: r == 0
=============================================================
[%  85|+ 231|-   2|T   0|S   0]: fs_symlink_dir
`fs_symlink_dir` failed: exit code 134
Output from process `fs_symlink_dir`:
Assertion failed in test/test-fs.c on line 1778: dent.type == UV_DIRENT_FILE
=============================================================
[%  86|+ 235|-   3|T   0|S   0]: fs_event_watch_dir_recursive
`fs_event_watch_dir_recursive` skipped
Output from process `fs_event_watch_dir_recursive`:
Recursive directory watching not supported on this platform.
=============================================================
[% 100|+ 270|-   3|T   0|S   1]: Done.
FAIL: test/run-tests
======================================================
1 of 1 test failed
Please report to https://github.com/libuv/libuv/issues
======================================================
make[1]: *** [check-TESTS] Error 1
make[1]: Leaving directory `/home/nan/libuv'
make: *** [check-am] Error 2 
~~~~

EDIT: showing a non-root run.

**Comment by saghul** [comment]
Tests must not be run as root.

**Comment by NanXiao** [comment]
@saghul : Oh! Why?

**Comment by saghul** [comment]
Because some tests drop privileges. Run them again with a clean slate and lets un know if you un into issues. Thanks!

**Comment by NanXiao** [comment]
@saghul:  When not root user, the test still failed:  

	[nan@localhost libuv]$ make check
	make  test/run-tests
	make[1]: Entering directory `/home/nan/libuv'
	  CC       test/test_run_tests-blackhole-server.o
	  CC       test/test_run_tests-dns-server.o
	  CC       test/test_run_tests-echo-server.o
	  CC       test/test_run_tests-run-tests.o
	  CC       test/test_run_tests-runner.o
	  CC       test/test_run_tests-test-active.o
	  CC       test/test_run_tests-test-async.o
	  CC       test/test_run_tests-test-async-null-cb.o
	  CC       test/test_run_tests-test-barrier.o
	  CC       test/test_run_tests-test-callback-order.o
	  CC       test/test_run_tests-test-callback-stack.o
	  CC       test/test_run_tests-test-close-fd.o
	  CC       test/test_run_tests-test-close-order.o
	  CC       test/test_run_tests-test-condvar.o
	  CC       test/test_run_tests-test-connection-fail.o
	  CC       test/test_run_tests-test-cwd-and-chdir.o
	  CC       test/test_run_tests-test-default-loop-close.o
	  CC       test/test_run_tests-test-delayed-accept.o
	  CC       test/test_run_tests-test-dlerror.o
	  CC       test/test_run_tests-test-embed.o
	  CC       test/test_run_tests-test-emfile.o
	  CC       test/test_run_tests-test-error.o
	  CC       test/test_run_tests-test-fail-always.o
	  CC       test/test_run_tests-test-fs-event.o
	  CC       test/test_run_tests-test-fs-poll.o
	  CC       test/test_run_tests-test-fs.o
	  CC       test/test_run_tests-test-get-currentexe.o
	  CC       test/test_run_tests-test-get-loadavg.o
	  CC       test/test_run_tests-test-get-memory.o
	test/test-get-memory.c: In function ‘run_test_get_memory’:
	test/test-get-memory.c:30:25: warning: ISO C90 does not support ‘long long’ [-Wlong-long]
			  (unsigned long long) free_mem,
							 ^
	test/test-get-memory.c:31:25: warning: ISO C90 does not support ‘long long’ [-Wlong-long]
			  (unsigned long long) total_mem);
							 ^
	test/test-get-memory.c:31:10: warning: ISO C90 does not support the ‘ll’ gnu_printf length modifier [-Wformat=]
			  (unsigned long long) total_mem);
			  ^
	test/test-get-memory.c:31:10: warning: ISO C90 does not support the ‘ll’ gnu_printf length modifier [-Wformat=]
	  CC       test/test_run_tests-test-getaddrinfo.o
	  CC       test/test_run_tests-test-getnameinfo.o
	  CC       test/test_run_tests-test-getsockname.o
	  CC       test/test_run_tests-test-handle-fileno.o
	  CC       test/test_run_tests-test-homedir.o
	  CC       test/test_run_tests-test-hrtime.o
	  CC       test/test_run_tests-test-idle.o
	  CC       test/test_run_tests-test-ip4-addr.o
	  CC       test/test_run_tests-test-ip6-addr.o
	  CC       test/test_run_tests-test-ipc-send-recv.o
	  CC       test/test_run_tests-test-ipc.o
	  CC       test/test_run_tests-test-loop-handles.o
	  CC       test/test_run_tests-test-loop-alive.o
	  CC       test/test_run_tests-test-loop-close.o
	  CC       test/test_run_tests-test-loop-stop.o
	  CC       test/test_run_tests-test-loop-time.o
	  CC       test/test_run_tests-test-loop-configure.o
	  CC       test/test_run_tests-test-multiple-listen.o
	  CC       test/test_run_tests-test-mutexes.o
	  CC       test/test_run_tests-test-osx-select.o
	  CC       test/test_run_tests-test-pass-always.o
	  CC       test/test_run_tests-test-ping-pong.o
	  CC       test/test_run_tests-test-pipe-bind-error.o
	  CC       test/test_run_tests-test-pipe-connect-error.o
	  CC       test/test_run_tests-test-pipe-connect-multiple.o
	  CC       test/test_run_tests-test-pipe-connect-prepare.o
	  CC       test/test_run_tests-test-pipe-getsockname.o
	  CC       test/test_run_tests-test-pipe-pending-instances.o
	  CC       test/test_run_tests-test-pipe-sendmsg.o
	  CC       test/test_run_tests-test-pipe-server-close.o
	  CC       test/test_run_tests-test-pipe-close-stdout-read-stdin.o
	  CC       test/test_run_tests-test-pipe-set-non-blocking.o
	  CC       test/test_run_tests-test-platform-output.o
	test/test-platform-output.c: In function ‘run_test_platform_output’:
	test/test-platform-output.c:50:59: warning: ISO C90 does not support ‘long long’ [-Wlong-long]
	   printf("uv_resident_set_memory: %llu\n", (unsigned long long) rss);
															   ^
	test/test-platform-output.c:50:3: warning: ISO C90 does not support the ‘ll’ gnu_printf length modifier [-Wformat=]
	   printf("uv_resident_set_memory: %llu\n", (unsigned long long) rss);
	   ^
	test/test-platform-output.c:65:25: warning: ISO C90 does not support ‘long long’ [-Wlong-long]
			  (unsigned long long) rusage.ru_utime.tv_sec,
							 ^
	test/test-platform-output.c:66:25: warning: ISO C90 does not support ‘long long’ [-Wlong-long]
			  (unsigned long long) rusage.ru_utime.tv_usec);
							 ^
	test/test-platform-output.c:66:10: warning: ISO C90 does not support the ‘ll’ gnu_printf length modifier [-Wformat=]
			  (unsigned long long) rusage.ru_utime.tv_usec);
			  ^
	test/test-platform-output.c:66:10: warning: ISO C90 does not support the ‘ll’ gnu_printf length modifier [-Wformat=]
	test/test-platform-output.c:68:25: warning: ISO C90 does not support ‘long long’ [-Wlong-long]
			  (unsigned long long) rusage.ru_stime.tv_sec,
							 ^
	test/test-platform-output.c:69:25: warning: ISO C90 does not support ‘long long’ [-Wlong-long]
			  (unsigned long long) rusage.ru_stime.tv_usec);
							 ^
	test/test-platform-output.c:69:10: warning: ISO C90 does not support the ‘ll’ gnu_printf length modifier [-Wformat=]
			  (unsigned long long) rusage.ru_stime.tv_usec);
			  ^
	test/test-platform-output.c:69:10: warning: ISO C90 does not support the ‘ll’ gnu_printf length modifier [-Wformat=]
	test/test-platform-output.c:78:50: warning: ISO C90 does not support ‘long long’ [-Wlong-long]
		 printf("  times.sys: %llu\n", (unsigned long long) cpus[i].cpu_times.sys);
													  ^
	test/test-platform-output.c:78:5: warning: ISO C90 does not support the ‘ll’ gnu_printf length modifier [-Wformat=]
		 printf("  times.sys: %llu\n", (unsigned long long) cpus[i].cpu_times.sys);
		 ^
	test/test-platform-output.c:80:27: warning: ISO C90 does not support ‘long long’ [-Wlong-long]
				(unsigned long long) cpus[i].cpu_times.user);
							   ^
	test/test-platform-output.c:80:12: warning: ISO C90 does not support the ‘ll’ gnu_printf length modifier [-Wformat=]
				(unsigned long long) cpus[i].cpu_times.user);
				^
	test/test-platform-output.c:82:27: warning: ISO C90 does not support ‘long long’ [-Wlong-long]
				(unsigned long long) cpus[i].cpu_times.idle);
							   ^
	test/test-platform-output.c:82:12: warning: ISO C90 does not support the ‘ll’ gnu_printf length modifier [-Wformat=]
				(unsigned long long) cpus[i].cpu_times.idle);
				^
	test/test-platform-output.c:83:51: warning: ISO C90 does not support ‘long long’ [-Wlong-long]
		 printf("  times.irq: %llu\n",  (unsigned long long) cpus[i].cpu_times.irq);
													   ^
	test/test-platform-output.c:83:5: warning: ISO C90 does not support the ‘ll’ gnu_printf length modifier [-Wformat=]
		 printf("  times.irq: %llu\n",  (unsigned long long) cpus[i].cpu_times.irq);
		 ^
	test/test-platform-output.c:85:27: warning: ISO C90 does not support ‘long long’ [-Wlong-long]
				(unsigned long long) cpus[i].cpu_times.nice);
							   ^
	test/test-platform-output.c:85:12: warning: ISO C90 does not support the ‘ll’ gnu_printf length modifier [-Wformat=]
				(unsigned long long) cpus[i].cpu_times.nice);
				^
	  CC       test/test_run_tests-test-poll-close.o
	  CC       test/test_run_tests-test-poll-close-doesnt-corrupt-stack.o
	test/test-poll-close-doesnt-corrupt-stack.c:114:0: warning: ISO C forbids an empty translation unit [-Wpedantic]
	 #endif  /* _WIN32 */
	 ^
	  CC       test/test_run_tests-test-poll-closesocket.o
	test/test-poll-closesocket.c:89:0: warning: ISO C forbids an empty translation unit [-Wpedantic]
	 #endif
	 ^
	  CC       test/test_run_tests-test-poll.o
	  CC       test/test_run_tests-test-process-title.o
	  CC       test/test_run_tests-test-ref.o
	  CC       test/test_run_tests-test-run-nowait.o
	  CC       test/test_run_tests-test-run-once.o
	  CC       test/test_run_tests-test-semaphore.o
	  CC       test/test_run_tests-test-shutdown-close.o
	  CC       test/test_run_tests-test-shutdown-eof.o
	  CC       test/test_run_tests-test-shutdown-twice.o
	  CC       test/test_run_tests-test-signal-multiple-loops.o
	  CC       test/test_run_tests-test-signal.o
	  CC       test/test_run_tests-test-socket-buffer-size.o
	  CC       test/test_run_tests-test-spawn.o
	  CC       test/test_run_tests-test-stdio-over-pipes.o
	  CC       test/test_run_tests-test-tcp-bind-error.o
	  CC       test/test_run_tests-test-tcp-bind6-error.o
	  CC       test/test_run_tests-test-tcp-close-accept.o
	  CC       test/test_run_tests-test-tcp-close-while-connecting.o
	  CC       test/test_run_tests-test-tcp-close.o
	  CC       test/test_run_tests-test-tcp-create-socket-early.o
	  CC       test/test_run_tests-test-tcp-connect-error-after-write.o
	  CC       test/test_run_tests-test-tcp-connect-error.o
	  CC       test/test_run_tests-test-tcp-connect-timeout.o
	  CC       test/test_run_tests-test-tcp-connect6-error.o
	  CC       test/test_run_tests-test-tcp-flags.o
	  CC       test/test_run_tests-test-tcp-open.o
	  CC       test/test_run_tests-test-tcp-read-stop.o
	  CC       test/test_run_tests-test-tcp-shutdown-after-write.o
	  CC       test/test_run_tests-test-tcp-unexpected-read.o
	  CC       test/test_run_tests-test-tcp-oob.o
	  CC       test/test_run_tests-test-tcp-write-to-half-open-connection.o
	  CC       test/test_run_tests-test-tcp-write-after-connect.o
	  CC       test/test_run_tests-test-tcp-writealot.o
	  CC       test/test_run_tests-test-tcp-write-fail.o
	  CC       test/test_run_tests-test-tcp-try-write.o
	  CC       test/test_run_tests-test-tcp-write-queue-order.o
	  CC       test/test_run_tests-test-thread-equal.o
	  CC       test/test_run_tests-test-thread.o
	  CC       test/test_run_tests-test-threadpool-cancel.o
	  CC       test/test_run_tests-test-threadpool.o
	  CC       test/test_run_tests-test-timer-again.o
	  CC       test/test_run_tests-test-timer-from-check.o
	  CC       test/test_run_tests-test-timer.o
	In file included from test/test-timer.c:23:0:
	test/test-timer.c: In function ‘run_test_timer_huge_timeout’:
	test/test-timer.c:233:59: warning: use of C99 long long integer constant [-Wlong-long]
	   ASSERT(0 == uv_timer_start(&huge_timer1, tiny_timer_cb, 0xffffffffffffLL, 0));
															   ^
	test/task.h:96:9: note: in definition of macro ‘ASSERT’
	   if (!(expr)) {                                          \
			 ^
	  CC       test/test_run_tests-test-tty.o
	  CC       test/test_run_tests-test-udp-bind.o
	  CC       test/test_run_tests-test-udp-create-socket-early.o
	  CC       test/test_run_tests-test-udp-dgram-too-big.o
	  CC       test/test_run_tests-test-udp-ipv6.o
	  CC       test/test_run_tests-test-udp-multicast-interface.o
	  CC       test/test_run_tests-test-udp-multicast-interface6.o
	  CC       test/test_run_tests-test-udp-multicast-join.o
	  CC       test/test_run_tests-test-udp-multicast-join6.o
	  CC       test/test_run_tests-test-udp-multicast-ttl.o
	  CC       test/test_run_tests-test-udp-open.o
	  CC       test/test_run_tests-test-udp-options.o
	  CC       test/test_run_tests-test-udp-send-and-recv.o
	  CC       test/test_run_tests-test-udp-send-immediate.o
	  CC       test/test_run_tests-test-udp-send-unreachable.o
	  CC       test/test_run_tests-test-udp-try-send.o
	  CC       test/test_run_tests-test-walk-handles.o
	  CC       test/test_run_tests-test-watcher-cross-stop.o
	  CC       test/test_run_tests-runner-unix.o
	  CCLD     test/run-tests
	make[1]: Leaving directory `/home/nan/libuv'
	make  check-TESTS
	make[1]: Entering directory `/home/nan/libuv'
	[%   0|+   0|-   0|T   0|S   0]: platform_output
	Output from process `platform_output`:
	uv_get_process_title: /home/nan/libuv/test/.libs/lt-run-tests
	uv_cwd: /home/nan/libuv
	uv_resident_set_memory: 778240
	uv_uptime: 4025.000000
	uv_getrusage:
	  user: 0 sec 2429 microsec
	  system: 0 sec 3644 microsec
	uv_cpu_info:
	  model: Intel(R) Core(TM) i5-4310U CPU @ 2.00GHz
	  speed: 2594
	  times.sys: 1272500
	  times.user: 6345200
	  times.idle: 30302300
	  times.irq: 1300
	  times.nice: 6900
	  model: Intel(R) Core(TM) i5-4310U CPU @ 2.00GHz
	  speed: 2594
	  times.sys: 1030900
	  times.user: 4833000
	  times.idle: 32717200
	  times.irq: 0
	  times.nice: 6900
	  model: Intel(R) Core(TM) i5-4310U CPU @ 2.00GHz
	  speed: 2594
	  times.sys: 990200
	  times.user: 4521100
	  times.idle: 33124800
	  times.irq: 0
	  times.nice: 5300
	  model: Intel(R) Core(TM) i5-4310U CPU @ 2.00GHz
	  speed: 2594
	  times.sys: 910000
	  times.user: 5148900
	  times.idle: 32086000
	  times.irq: 0
	  times.nice: 7000
	uv_interface_addresses:
	  name: lo
	  internal: 1
	  physical address: 00:00:00:00:00:00
	  address: 127.0.0.1
	  netmask: 255.0.0.0
	  name: enp0s3
	  internal: 0
	  physical address: 08:00:27:7d:e3:ea
	  address: 16.187.246.152
	  netmask: 255.255.252.0
	  name: virbr0
	  internal: 0
	  physical address: fe:54:00:d1:7f:7e
	  address: 192.168.122.1
	  netmask: 255.255.255.0
	  name: lo
	  internal: 1
	  physical address: 00:00:00:00:00:00
	  address: ::1
	  netmask: ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff
	  name: enp0s3
	  internal: 0
	  physical address: 08:00:27:7d:e3:ea
	  address: fe80::a00:27ff:fe7d:e3ea
	  netmask: ffff:ffff:ffff:ffff::
	  name: virbr0
	  internal: 0
	  physical address: fe:54:00:d1:7f:7e
	  address: fe80::fc54:ff:fedf:fbb7
	  netmask: ffff:ffff:ffff:ffff::
	  name: vnet0
	  internal: 0
	  physical address: fe:54:00:d1:7f:7e
	  address: fe80::fc54:ff:fed1:7f7e
	  netmask: ffff:ffff:ffff:ffff::
	=============================================================
	[%  80|+ 221|-   0|T   0|S   0]: fs_async_dir
	`fs_async_dir` failed: exit code 134
	Output from process `fs_async_dir`:
	Assertion failed in test/test-fs.c on line 440: dent.type == UV_DIRENT_FILE
	=============================================================
	[%  82|+ 224|-   1|T   0|S   0]: fs_access
	`fs_access` failed: exit code 134
	Output from process `fs_access`:
	Assertion failed in test/test-fs.c on line 1204: r == 0
	=============================================================
	[%  85|+ 231|-   2|T   0|S   0]: fs_symlink_dir
	`fs_symlink_dir` failed: exit code 134
	Output from process `fs_symlink_dir`:
	Assertion failed in test/test-fs.c on line 1778: dent.type == UV_DIRENT_FILE
	=============================================================
	[%  86|+ 235|-   3|T   0|S   0]: fs_event_watch_dir_recursive
	`fs_event_watch_dir_recursive` skipped
	Output from process `fs_event_watch_dir_recursive`:
	Recursive directory watching not supported on this platform.
	=============================================================
	[% 100|+ 270|-   3|T   0|S   1]: Done.
	FAIL: test/run-tests
	======================================================
	1 of 1 test failed
	Please report to https://github.com/libuv/libuv/issues
	======================================================
	make[1]: *** [check-TESTS] Error 1
	make[1]: Leaving directory `/home/nan/libuv'
	make: *** [check-am] Error 2

**Comment by whitlockjc** [comment]
I can confirm this issue.  I built without a privileged user on CentOS 7, RHEL 7 equivalent, and the same tests fail:

```
=============================================================
[%  80|+ 219|-   0|T   0|S   0]: fs_async_dir
`fs_async_dir` failed: exit code 134
Output from process `fs_async_dir`:
Assertion failed in test/test-fs.c on line 440: dent.type == UV_DIRENT_FILE
=============================================================
[%  81|+ 222|-   1|T   0|S   0]: fs_access
`fs_access` failed: exit code 134
Output from process `fs_access`:
Assertion failed in test/test-fs.c on line 1204: r == 0
=============================================================
[%  84|+ 229|-   2|T   0|S   0]: fs_symlink_dir
`fs_symlink_dir` failed: exit code 134
Output from process `fs_symlink_dir`:
Assertion failed in test/test-fs.c on line 1778: dent.type == UV_DIRENT_FILE
=============================================================
[%  86|+ 233|-   3|T   0|S   0]: fs_event_watch_dir_recursive
`fs_event_watch_dir_recursive` skipped
Output from process `fs_event_watch_dir_recursive`:
Recursive directory watching not supported on this platform.
=============================================================
[% 100|+ 268|-   3|T   0|S   1]: Done.
```

I'll take a peek.

**Comment by whitlockjc** [comment]
`fs_async_dir` and `fs_symlink_dir` are returning `UV_DIRENT_UNKNOWN` instead of `UV_DIRENT_FILE`.  I'm trying to see now if that is on purpose or not.  I'll look into `fs_access` after I figure the other two out since they seem to be related.

**Comment by saghul** [comment]
What filesystem is that?

**Comment by whitlockjc** [comment]
xfs

**Comment by saghul** [comment]
It's not supported on XFS:

~~~~
Currently, only some file systems (among them: Btrfs, ext2, ext3, and ext4) have full support for returning the file type in d_type. All applications must properly handle a return of DT_UNKNOWN. 
~~~~

http://linux.die.net/man/2/getdents

**Comment by whitlockjc** [comment]
Yup, I found the same results.  So...do you mind the test checking for both?  If so, I'll submit a PR.

**Comment by saghul** [comment]
Hum. I added stricter checks because a bug went unnoticed on Windows because of it. I think I'd be ok with a PR which checks for both on Unices.

**Comment by whitlockjc** [comment]
That would work.  It would be slick if we could have `HAVE_DIRENT_TYPES` set better such that it is known at test runtime if it should do the proper check or not because like Windows, I can see this allowing bugs to escape.

**Comment by whitlockjc** [comment]
#296 ring a bell?

**Comment by saghul** [comment]
The problem is that even if `HAVE_DIRENT_TYPES` is defined, it won't work for all file systems, at runtime.

**Comment by whitlockjc** [comment]
No, I get that.  :)  I'm saying I wish there was a way for that to be smarter but it doesn't make any sense really the more I think about it.  This is really just a test thing.  I'll submit a PR.

**Comment by saghul** [comment]
> #296 ring a bell?

Time to revisit, I guess :-)

**Comment by whitlockjc** [comment]
Nah man, I just saw it when looking around and wanted to link to things just in case it became important later.

**Comment by whitlockjc** [comment]
Once I typed that out, I realized my intent wasn't clear based on my wording so I deleted it.  I just saw that #296 reported similar failures and you had a similar discussion there as well.

**Comment by whitlockjc** [comment]
I've created a PR and tried to isolate the code that does the checks.  There were a few places where we were doing the dirent type check and now I've put that into a function so that we can fix it once and all references to it will be fixed as well.  I also decided to not check for `UV_UNKNOWN` on Windows, as you requested, and OS X as well since it is known to fully support getting the file type.  As mentioned in the commit message, if we know of other environments where it is getting the file type is guaranteed, we can update one place now instead of 3+.
